### PR TITLE
MON-2973: pkg/manifests: Allow configuring secrets in alertmanager (UWM)

### PR DESCRIPTION
**Pull request by slashpai**
Related-to #[MON-2798](https://issues.redhat.com//browse/MON-2798)

Signed-off-by: Jayapriya Pai <janantha@redhat.com>

* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by slashpai** [comment]
https://github.com/openshift/cluster-monitoring-operator/pull/1882 is already added for platform

**Review by JoaoBraveCoding — APPROVED** [review]
From my side it's a lgtm 👍

File: pkg/manifests/manifests.go [hunk 1]
```diff
@@ -460,6 +460,10 @@ func (f *Factory) AlertmanagerUserWorkload(trustedCABundleCM *v1.ConfigMap) (*mo
 		a.Spec.Resources = *alertmanagerConfig.Resources
 	}
 
+	if alertmanagerConfig.Secrets != nil {
+		a.Spec.Secrets = append(a.Spec.Secrets, alertmanagerConfig.Secrets...)
+	}
```
Comment (simonpasquier) [inline pkg/manifests/manifests.go:465]: I think that we can skip checking if `alertmanagerConfig.Secrets` is nil.
```suggestion
	a.Spec.Secrets = append(a.Spec.Secrets, alertmanagerConfig.Secrets...)
```

**Review by simonpasquier — COMMENTED** [review]
1. I'd add an end-to-end test.
2. If a secret is missing, the Alertmanager rollout will fail. Should CMO check that all secrets exist? It could be a follow-up PR.

**Comment by slashpai** [comment]
> 1. I'd add an end-to-end test.
> 2. If a secret is missing, the Alertmanager rollout will fail. Should CMO check that all secrets exist? It could be a follow-up PR.

sure I will add e2e test

We check if all secret exists here
https://github.com/openshift/cluster-monitoring-operator/pull/1884/files#diff-2d639109b98574a4d78f5d7f9f92e79af553db04b91e4faaa5ed6e9943f197bfR95-R106. Same as implemented for platform(https://github.com/openshift/cluster-monitoring-operator/pull/1882). Any change needed here?

**Comment by simonpasquier** [comment]
> We check if all secret exists here
> https://github.com/openshift/cluster-monitoring-operator/pull/1884/files#diff-2d639109b98574a4d78f5d7f9f92e79af553db04b91e4faaa5ed6e9943f197bfR95-R106. Same as implemented for platform(#1882). Any change needed here?

Ah I missed it, :+1:

**Comment by juzhao** [comment]
> > 1. I'd add an end-to-end test.
> > 2. If a secret is missing, the Alertmanager rollout will fail. Should CMO check that all secrets exist? It could be a follow-up PR.
> 
> sure I will add e2e test
> 
> We check if all secret exists here https://github.com/openshift/cluster-monitoring-operator/pull/1884/files#diff-2d639109b98574a4d78f5d7f9f92e79af553db04b91e4faaa5ed6e9943f197bfR95-R106. Same as implemented for platform(#1882). Any change needed here?

if the secret not exist, it will throw out error in CMO
```
E0214 08:45:38.336450       1 operator.go:559] sync "openshift-monitoring/cluster-monitoring-config" failed: cluster monitoring update failed (reason: UpdatingAlertmanagerUserWorkloadFailed)
W0214 08:55:43.602238       1 tasks.go:72] task 6 of 14: Updating Alertmanager-user-workload failed: failed to find Alertmanager secret "test-secret-uwm": waiting for secret openshift-user-workload-monitoring/test-secret-uwm: secrets "test-secret-uwm" not found
```
and monitoring would be degraded
```
$ oc get co monitoring
NAME         VERSION                                                   AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
monitoring   4.13.0-0.ci.test-2023-02-14-065113-ci-ln-s6798tt-latest   False       True          True       14m     failed to find Alertmanager secret "test-secret-uwm": waiting for secret openshift-user-workload-monitoring/test-secret-uwm: secrets "test-secret-uwm" not found
```

File: pkg/manifests/manifests.go (hunk 1 above)
Comment (jan--f) [reply pkg/manifests/manifests.go:465]: +1 though if this check is dropped, the same should be added to `AlertmanagerMain()` at  https://github.com/openshift/cluster-monitoring-operator/blob/d4d3ad1812c797080befa27592717e0c53ecf81a/pkg/manifests/manifests.go#L585

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Comment by jan--f** [comment]
@bburt-rh another PR for your esteemed attention :)

**Comment by slashpai** [comment]
/label docs-approved
Since https://issues.redhat.com/browse/RHDEVDOCS-4842 is already created for docs

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Pull request merged**